src/middlewares/__tests__/createInsightsMiddleware.ts

```diff
@@ -127,22 +127,96 @@ describe('insights', () => {
         },
         widgetType: 'ais.hits',
       });
-      expect(analytics.viewedObjectIDs).toHaveBeenCalledTimes(0);
-
-      // But, the library hasn't been loaded yet, so the event stays in the queue.
-      expect(insightsClient.queue[insightsClient.queue.length - 1]).toEqual([
-        'viewedObjectIDs',
-        { eventName: 'Hits Viewed', index: '', objectIDs: ['1', '2'] },
-      ]);
 
       // When the library is loaded later, it consumes the queue and sends the event.
+      expect(() => {
+        libraryLoadedAndProcessQueue();
+      }).not.toThrow(
+        "Before calling any methods on the analytics, you first need to call 'setUserToken' function or include 'userToken' in the event payload."
+      );
+    });
+
+    it('does not throw without userToken in UMD with the library loaded before the event', () => {
+      const {
+        insightsClient,
+        libraryLoadedAndProcessQueue,
+        instantSearchInstance,
+      } = createUmdTestEnvironment();
+
+      const middleware = createInsightsMiddleware({
+        insightsClient,
+        insightsInitParams: { useCookie: false },
+      })({ instantSearchInstance });
+      middleware.subscribe();
+
       libraryLoadedAndProcessQueue();
-      expect(analytics.viewedObjectIDs).toHaveBeenCalledTimes(1);
-      expect(analytics.viewedObjectIDs).toHaveBeenCalledWith({
-        eventName: 'Hits Viewed',
-        index: '',
-        objectIDs: ['1', '2'],
-      });
+
+      expect(() => {
+        // It tries to send an event.
+        instantSearchInstance.sendEventToInsights({
```

Comment:
It's out of scope for this PR but we're testing with internals here. It would be less fragile to actually mount a widget and interact with it to trigger an event.

Reply:
Yes, I agree. I'll convert some unit tests to integration tests in the near future.